### ISSUE TITLE
Local search too specific

### ISSUE THREAD

**Issue by r0ckarong** [doxygen/doxygen#8505]
**Describe the bug**
I have enabled SEARCHENGINE only and it sort of works but the search terms have to be nearly perfect with what I am trying to find. In the attached example there are two enums `ABCD` and `A_LONGER_STRING`.

When searching: 

- A - Returns both as results
- _LONGER / LONGER - No Matches
- STRING - No Matches
- A_LONGER - Matches the second Enum correctly.

Am I misunderstanding what this is supposed to achieve? I can not grasp how a search would only show things that you already know.

**Expected behavior**
I can search for fuzzy terms "containing" what I might be looking for and get close matches that I can choose from.

**Screenshots**
![image](https://user-images.githubusercontent.com/4920542/115239116-b0aab380-a11e-11eb-8d89-b93434d103bb.png)
![image](https://user-images.githubusercontent.com/4920542/115239160-b7d1c180-a11e-11eb-9a55-176e5b076e18.png)
![image](https://user-images.githubusercontent.com/4920542/115239189-bf916600-a11e-11eb-801d-e7ef51c98b07.png)

**To Reproduce**

[doxygen_search_debug.zip](https://github.com/doxygen/doxygen/files/6336186/doxygen_search_debug.zip)

**Version**
1.9.1 on 

Windows 10 64bit 
Major  Minor  Build  Revision
-----  -----  -----  --------
10     0      19042  0

**Comment by albert-github** [comment]
The default search engine is a simple search engine starting to search from the beginning of the input string, so when having `A` it finds both items but when starting at e.g. `_` nothing is found as there are no searchable entities starting with `_`.
This is by design, one can always use other search engines (see settings e.g. `EXTERNAL_SEARCH`, `SERVER_BASED_SEARCH`).

So the problem is an interpretation of the possibilities  and thus usage.

Note: the provided Dxoyfile has a small flaw (probably due to stripping the own original Doxyfile):
```
OUTPUT_DIRECTORY       = build\
CREATE_SUBDIRS         = NO
```
should read
```
OUTPUT_DIRECTORY       = build
CREATE_SUBDIRS         = NO
```
otherwise the 2 lines are concatenated.

**Comment by r0ckarong** [comment]
> So the problem is an interpretation of the possibilities and thus usage.

Thanks for clarifying this. I thought so but it didn't become clear from the official docs what the capatbilities are.

Thanks for spotting the mistake in the Doxyfile. That is on me :)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8500 with a small explanation for the documentation.

**Comment by albert-github** [comment]
Updated documentation has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
